### PR TITLE
Ensure selected item number in Sequence browser remains valid

### PR DESCRIPTION
**Pull request by lassoan**
Two small fixes in keeping selected item number in sequence browser valid when browsing/adding/deleting items in the master sequence node.

These are not critical fixes, not required to be included in 5.6.1 patch release.

### CODE REVIEW

File: Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx [hunk 1]
```diff
@@ -501,7 +501,7 @@ std::string vtkMRMLSequenceBrowserNode::SetAndObserveMasterSequenceNodeID(const
     this->RemoveAllSequenceNodes();
     // Master is the first element in the postfixes vector.
     std::string rolePostfix = this->AddSynchronizedSequenceNodeID(sequenceNodeID);
-    this->SetSelectedItemNumber(0);
+    this->SetSelectedItemNumber(this->GetNumberOfItems() > 0 ? 0 : -1);
```
[the remark this answers is not in the document]
Comment (jcfr) [reply Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:504]: May be worth adding a comment explaining the rational for mapping number of items to  selected item "index"

Is the doxygen associated with `` correct ? It current mention `/// Get/Set selected bundle index` but the term `bundle` is mentioned nowhere else.

File: Modules/Loadable/Sequences/Testing/Cxx/vtkMRMLSequenceBrowserNodeTest1.cxx [hunk 2]
```diff
@@ -24,18 +24,22 @@
...
 double valueForIndex(int i)
 {
   return i * 1234.567890;
 }
 
-int vtkMRMLSequenceBrowserNodeTest1(int vtkNotUsed(argc), char* vtkNotUsed(argv)[])
+int TestIndexFormatting()
 {
-  vtkNew<vtkMRMLScene> scene;
+    vtkNew<vtkMRMLScene> scene;
```
[the remark this answers is not in the document]
Comment (lassoan) [reply Modules/Loadable/Sequences/Testing/Cxx/vtkMRMLSequenceBrowserNodeTest1.cxx:42]: Good catchm I'll fix.

File: Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lassoan) [reply Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:504]: Good points, I'll fix these as suggested.

File: Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx [hunk 3]
```diff
@@ -1088,6 +1097,26 @@ void vtkMRMLSequenceBrowserNode::ProcessMRMLEvents(vtkObject* caller, unsigned l
     }
   else if (vtkMRMLSequenceNode::SafeDownCast(modifiedNode))
     {
+    if (this->SelectedItemNumber < 0)
+      {
+      // There was no valid selected item number. If the master sequence node now contains an item
+      // then automatically select it.
+      // This is consistent with automatically setting selected item number to INVALID_ITEM_NUMBER when all items are removed.
```
Comment (jcfr) [inline Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:1104]: Looking at `RemoveAllSequenceNodes`, I can see that `INVALID_ITEM_NUMBER` is set. But when I look at `RemoveSynchronizedSequenceNode`, I do not see when/where `SetSelectedItemNumber` is called with `INVALID_ITEM_NUMBER`

Comment (lassoan) [reply Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:1104]: It is implemented in `vtkMRMLSequenceBrowserNode::ProcessMRMLEvents`. When the master sequence node is changed then `SelectedItemNumber` may be updated.

File: Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx [hunk 4]
```diff
@@ -49,13 +49,17 @@
...
 
-// First reference is the master sequence node, subsequent references are the synchronized sequence nodes
-static const char* SEQUENCE_NODE_REFERENCE_ROLE_BASE = "sequenceNodeRef"; // Old: rootNodeRef
-// TODO: Change this to "proxyNodeRef", but need to maintain backwards-compatibility with "dataNodeRef"
-static const char* PROXY_NODE_REFERENCE_ROLE_BASE = "dataNodeRef";
+namespace
+{
+  // First reference is the master sequence node, subsequent references are the synchronized sequence nodes
+  const char* SEQUENCE_NODE_REFERENCE_ROLE_BASE = "sequenceNodeRef"; // Old: rootNodeRef
+  // TODO: Change this to "proxyNodeRef", but need to maintain backwards-compatibility with "dataNodeRef"
```
Comment (jcfr) [inline Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:56]: Should an issue be created ? Or should the comment be changed to:

```
  // Ideally this should be changed to "proxyNodeRef" but we need to maintain it
  // for backwards-compatibility with "dataNodeRef"
```

Comment (lassoan) [reply Modules/Loadable/Sequences/MRML/vtkMRMLSequenceBrowserNode.cxx:56]: Good point, I'll use the text you've suggested.

**Comment by lassoan** [comment]
@jcfr I've addressed your comments, please have a look and approve if it looks good to you.

**Review by jcfr — APPROVED** [review]

**Pull request merged**